client/components/checklist/task.js

```diff
@@ -172,50 +154,50 @@ class Task extends PureComponent {
 					warning: isWarning,
 					'is-completed': completed,
 					'is-in-progress': inProgress,
-					'has-actionlink': hasActionlink,
-					'is-collapsed': this.isTaskCollapsed(),
+					'is-unexpandable': ! isExpandable,
+					'is-collapsed': collapsed,
 				} ) }
 			>
-				<div className="checklist__task-primary">
+				<div className="checklist__task-wrapper">
 					<h3 className="checklist__task-title">
-						{ completed ? (
-							completedTitle || title
-						) : (
+						{ isExpandable ? (
 							<Button
 								borderless
-								className="checklist__task-title-link"
-								href={ href }
-								onClick={ this.onTaskClick }
-								target={ target }
+								className="checklist__task-title-button"
+								onClick={ this.props.onTaskClick }
 							>
-								{ title }
+								{ completed ? completedTitle : title }
+								<Gridicon icon="chevron-up" className="checklist__toggle-affordance" />
 							</Button>
+						) : (
+							completedTitle
 						) }
 					</h3>
-					<p className="checklist__task-description">{ description }</p>
-					{ completedDescription && (
-						<p className="checklist__task-completed-description">{ completedDescription }</p>
-					) }
-					{ duration && (
-						<small className="checklist__task-duration">
-							{ translate( 'Estimated time:' ) } { duration }
-						</small>
-					) }
-				</div>
-				<div className="checklist__task-secondary">
-					<Button
-						className="checklist__task-action"
-						href={ href }
-						onClick={ onClick }
-						primary={ buttonPrimary }
-						target={ target }
-					>
-						{ taskActionButtonText }
-					</Button>
-					{ duration && (
-						<small className="checklist__task-duration">
-							{ translate( 'Estimated time:' ) } { duration }
-						</small>
+
+					{ ! collapsed && (
+						<div className="checklist__task-content">
+							<p className="checklist__task-description">{ description }</p>
+
+							<div className="checklist__task-action-duration-wrapper">
```

Comment:
Nit; this class could be `-actions` or something like that, since it wraps duration _and_ action, as well as possible future "skip" button.

Not a blocker.